### PR TITLE
Fix/serializer

### PR DESCRIPTION
**Pull request by RoyAyo**

### CODE REVIEW

**Review by georgeijidola — CHANGES_REQUESTED** [review]

File: src/deserialize/index.ts [hunk 1]
```diff
@@ -19,46 +18,46 @@ export interface DeserializeFunctions {
...
-export function buildDeserializerForModelStandalone() {
-  return {
-    create: async (data, options: DeserializerOptions = {}) => {
-      try {
-        return new Deserializer(options).deserialize(data);
-      } catch (err) {
-        return data;
-      }
-    },
+// export function buildDeserializerForModelStandalone() {
```
[the remark this answers is not in the document]
Comment (RoyAyo) [reply src/deserialize/index.ts:25]: Yes, we may need to rebuild the function later and may use it.

File: src/deserialize/index.ts [hunk 2]
```diff
@@ -19,46 +18,46 @@ export interface DeserializeFunctions {
...
   scaffold: Scaffold,
   modelName: string
 ): DeserializeFunctions {
   return {
-    create: async (data, options: DeserializerOptions = {}) => {
+    create: async (data: any) => {
       try {
-        return new Deserializer(options).deserialize(data);
+        return scaffold.serializer.deserialize(modelName, data);
       } catch (err) {
```
[the remark this answers is not in the document]
Comment (RoyAyo) [reply src/deserialize/index.ts:53]: The data cold not be deserialized, return the data in the appropriate format.

File: src/index.ts [hunk 3]
```diff
@@ -47,9 +48,9 @@ import { IAssociation } from "./sequelize/types";
  * @param {ScaffoldModel} model The Scaffold Model to use for validation, attributes, relationships, etc
  * @returns {ModelFunctionsCollection<ParseFunctions>}
  */
-export function Parse(model: ScaffoldModel) {
-  return buildParserForModelStandalone(model);
-}
+// export function Parse(model: ScaffoldModel) {
```
[the remark this answers is not in the document]
Comment (RoyAyo) [reply src/index.ts:51]: Yes, sane with the previous, these functions are builds to be separated from Scaffold. It cannot work now, but may be needed soon

File: src/deserialize/test.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
+/* eslint-disable no-unused-vars */
+import { Deserializer, DeserializerOptions } from "jsonapi-serializer";
+import { Scaffold } from "..";
+import { JSONObject } from "../types";
+
```
[the remark this answers is not in the document]
Comment (RoyAyo) [reply src/deserialize/test.ts:5]: No, it's the previous deserializer implementation and may be deleted actually. Only index.ts is used

**Review by georgeijidola — APPROVED** [review]

**Pull request merged**